### ISSUE TITLE
Recommend use of `--enable-optimizations`?

### ISSUE THREAD

**Issue by hauntsaninja** [pyenv/pyenv#2577]
### Prerequisite
* [x] Make sure your problem is not listed in [the common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems).
* [x] Make sure no duplicated issue has already been reported in [the pyenv issues](https://github.com/pyenv/pyenv/issues). You should look for closed issues, too.
* [x] The rest of the prerequisites (they're less applicable)

Several users of pyenv are surprised to find out that their Python could be ~30% faster by using `--enable-optimizations`.

I don't see any discussion in this repo about whether `--enable-optimizations` should be the default (possibly with an appropriate default `PROFILE_TASK`). Assuming you don't want to change the default, it might be nice to at least mention this as a thing users could do somewhere in the README.

These are the env vars I set when building with pyenv:
```
export PYTHON_CONFIGURE_OPTS='--enable-optimizations --with-lto'
export PROFILE_TASK='-m test.regrtest --pgo -j0'
export PYTHON_CFLAGS='-march=native -mtune=native'
```

**Comment by native-api** [comment]
* `--enable-optimizations` is PGO and it makes the build take a few times longer 'cuz it needs to: 1) make an instrumented build 2) run the training code (usually, the test suite) 3) make the complete build using the trained model.

As you can see, making a sensible optimization here requires knowledge of what kind of code you're going to be running.
So it's a niche use case for dedicated servers.

Plus, for most of the users, performance is not critical, and Python itself is deliberately designed to sacrifice code performance for programmer's performance.

To summarize, this looks like too taxing for regular use to be the default. But it can probably be described in the wiki.

**Comment by hauntsaninja** [comment]
Thanks for the reply (and for pyenv itself)! :-)

I wouldn't say it's niche... In my experience, you get a pretty consistent speedup just by using the default profile task (of running some unit tests) and the marginal benefit of using an application-specific task for the profile task is low to zero. I agree that for most users performance is not critical, but people seem to be quite happy about Python 3.11, and this is the same ballpark speedup.

Anyway, I added a section to the wiki over here: https://github.com/pyenv/pyenv/wiki#how-to-build-cpython-for-maximum-performance. I also opened a PR https://github.com/pyenv/pyenv/pull/2579 to briefly mention this in the README, since that is by far the most discoverable location for users. Let me know what you think
